app/models/description.rb

```diff
@@ -400,25 +400,46 @@ def group_users(table)
   # Array of ids.  Caches result.
   def group_user_ids(table)
     @group_user_ids ||= {}
-    @group_user_ids[table] ||= self.class.connection.select_values(%(
-      SELECT DISTINCT u.user_id FROM #{table} t, user_groups_users u
-      WHERE t.#{type_tag}_id = #{id}
-        AND t.user_group_id = u.user_group_id
-      ORDER BY u.user_id ASC
-    )).map(&:to_i)
+    select_manager = arel_select_group_user_ids(table)
+    @group_user_ids[table] ||= self.class.connection.
+                               select_values(select_manager.to_sql).map(&:to_i)
   end
```

Comment:
I wonder if we might not be better off using AR here, too.  It will instantiate a bunch of UserGroups and Users that won't ever be used, but maybe the performance hit would be minimal, a reasonable price to pay for reducing this entire mess to simply:

  old --> name_description.group_user_ids(admins_join_table)
  new --> name_description.admin_groups.map(&:users).flatten.map(&:id)

  old --> name_description.group_ids(admins_join_table)
  new --> name_description.admin_groups.ids

And then we remove these two methods (and all their ABC-violating helpers) completely!

Reply:
Seems like this would mean removing three methods here, or am i misreading.
`#group_users(table)`, #group_user_ids(table)`, `#group_ids(table)`

Would this also be a use case for that `deep_pluck` gem?

Reply:
Yes, group_users, too, thanks.  Is there also a groups?

Reply:
Nope there's no `groups`. 
But are you suggesting above that the three methods should be removed, and MO should use `admin_groups.ids` in the codebase instead? Or that these methods should simply get their values directly from `admin_groups` etc. Sorry i'm confused!